python/src/functions/subrecipient_treasury_report_gen.py

```diff
@@ -11,11 +29,189 @@ def handle(event: Dict[str, Any], context: Context):
         {
             organization: <all fields in organization object>,
             user: <id and email fields of the user object>,
+            outputTemplateId: <id for the output template to use>
         }
         context: Lambda context
+
+    This function should:
+    1. Parse necessary inputs from the event
+    2. Download a file of recent subrecipients from S3 to put into the output template
+    3. Download the output template itself from S3
+    4. Iterate through recent subrecipients and put their information into the output template
+    5. Upload the output template, in both xlsx and csv formats, to S3
     """
+    structlog.contextvars.bind_contextvars(
+        lambda_event={"subrecipient_step_function": event}
+    )
     logger = get_logger()
-    # TODO take a closer look at the below when implementing function
-    if (event is None or context is None):
+    logger.info("received new invocation event from step function")
+    if not event or not context:
         logger.exception("Missing event or context")
         return
+
+    organization_id = ...
+    reporting_period_id = ...
+    output_template_id = ...
+    user_id = ...
+
+    try:
+        reporting_period_id = event["organization"]["preferences"][
+            "current_reporting_period_id"
+        ]
+        organization_id = event["organization"]["id"]
+        output_template_id = event["outputTemplateId"]
+        user_id = event["user"]["id"]
+    except KeyError as e:
+        logger.exception(
+            f"Exception getting reporting period or organization id from event -- missing field: {e}"
+        )
+        return
+
+    s3_client: S3Client = boto3.client("s3")
+
+    with tempfile.NamedTemporaryFile() as recent_subrecipients_file:
+        with structlog.contextvars.bound_contextvars(
+            subrecipients_filename=recent_subrecipients_file.name
+        ):
+            download_s3_object(
+                s3_client,
+                BUCKET_NAME,
+                f"/{organization_id}/{reporting_period_id}/subrecipients",
```

Comment:
@laurelkorwin should this file be ending in `.json` here? I wasn't sure if there was an extension to this file or not.